### PR TITLE
[exporter/kafkaexporter] Add support for configuring SASL handshake version

### PR DESCRIPTION
**Pull request by eduojeda**
**Description:**
Adds support for configuring SASL handshake version.

The new parameter defaults to 0 just like Sarama, so the change should be backwards compatible.

**Link to tracking Issue:** #21074 

**Testing:**
- Added a test for the check that the SASL version parameter is within the acceptable range
- Built a custom collector and tested that the version setting works as expected.

**Documentation:**
- Updated the README file to document the new SASL configuration parameter.

### CODE REVIEW

File: exporter/kafkaexporter/authentication.go
Comment (atoulme) [inline exporter/kafkaexporter/authentication.go:142]: Please add a validation check in Validate()

File: exporter/kafkaexporter/authentication_test.go
Comment (atoulme) [inline exporter/kafkaexporter/authentication_test.go:127]: I don't see a test using version 1?

**Comment by atoulme** [comment]
Please add a CHANGELOG entry.

File: exporter/kafkaexporter/authentication.go
Comment (eduojeda) [reply exporter/kafkaexporter/authentication.go:142]: I'm happy to add the check, but I'm a little confused about this. I don't see corresponding validation checks for most of the other keys (e.g. `sasl.mechanism`), so what makes this new one different? I was just following the patterns that I see in the existing codebase. There are only two checks in `Validate()` right now and it's not obvious to me what the criteria is.

File: exporter/kafkaexporter/authentication_test.go
Comment (eduojeda) [reply exporter/kafkaexporter/authentication_test.go]: Good shout, I introduced the test

File: exporter/kafkaexporter/authentication.go
Comment (eduojeda) [reply exporter/kafkaexporter/authentication.go:142]: Fair enough, I added validation checks for the other `Authentication.SASL` keys as well

Comment (atoulme) [reply exporter/kafkaexporter/authentication.go:142]: Thank you ;)

**Comment by eduojeda** [comment]
@atoulme I'm sorry, I'm a bit lost as to how to proceed here. Is there anything else I need to do or is this PR just waiting for a codeowner review now?

**Comment by atoulme** [comment]
Please rebase to fix the build.

**Comment by eduojeda** [comment]
@atoulme I've been notified that this branch has been marked as stale. As far as I can tell everything is ready for the merge. I've just re-synced with the upstream as well. Would it be possible to give it a final check?

**Review by MovieStoreGuy — COMMENTED** [review]
A few comments from me, I think this should also update the kafka receiver as well.

File: exporter/kafkaexporter/config_test.go
Comment (eduojeda) [reply exporter/kafkaexporter/config_test.go]: Good shout, I corrected a couple other instances of this as well

**Comment by eduojeda** [comment]
@MovieStoreGuy I've addressed your comments, is there anything else that I should do to before we can merge the PR?

**Review by MovieStoreGuy — APPROVED** [review]

**Pull request merged**
